api/common/interfaces.go

```diff
@@ -647,91 +647,152 @@ func getSubscribers(s Serverer, params map[string]interface{}) map[string]interf
 		return respPacking(INVALID_PARAMS, "topic should be a string")
 	}
 
-	bucket, ok := params["bucket"].(float64)
+	var bucket float64
+	if _, ok := params["bucket"]; ok {
+		bucket, ok = params["bucket"].(float64)
+		if !ok {
+			return respPacking(INVALID_PARAMS, "bucket should be a float64")
+		}
+	}
+
+	subscriber, ok := params["subscriber"].(string)
 	if !ok {
-		return respPacking(INVALID_PARAMS, "bucket should be a string")
+		return respPacking(INVALID_PARAMS, "subscriber should be a string")
+	}
+
+	_, pubKey, identifier, err := address.ParseClientAddress(subscriber)
+	if err != nil {
+		return respPacking(INTERNAL_ERROR, err.Error())
 	}
 
-	subscribers, err := chain.DefaultLedger.Store.GetSubscribers(topic, uint32(bucket))
+	meta, expiresAt, err := chain.DefaultLedger.Store.GetSubscription(topic, uint32(bucket), pubKey, identifier)
 	if err != nil {
 		return respPacking(INTERNAL_ERROR, err.Error())
 	}
-	return respPacking(SUCCESS, subscribers)
+	return respPacking(SUCCESS, struct {
+		Meta      string `json:"meta"`
+		ExpiresAt uint32 `json:"expiresAt"`
+	}{
+		meta,
+		expiresAt,
+	})
 }
 
-// getAsset get subscribers by topic
-// params: {"assetid":<id>}
+// getSubscribers get subscribers by topic
+// params: {"topic":<topic>, "bucket":<bucket>, "offset":<offset>, "limit":<limit>}
 // return: {"resultOrData":<result>|<error data>, "error":<errcode>}
-func getAsset(s Serverer, params map[string]interface{}) map[string]interface{} {
-	if len(params) < 1 {
-		return respPacking(INVALID_PARAMS, "length of params is less than 1")
+func getSubscribers(s Serverer, params map[string]interface{}) map[string]interface{} {
+	if len(params) < 2 {
+		return respPacking(INVALID_PARAMS, "length of params is less than 2")
 	}
 
-	str, ok := params["assetid"].(string)
+	topic, ok := params["topic"].(string)
 	if !ok {
-		return respPacking(INVALID_PARAMS, "asset ID should be a string")
+		return respPacking(INVALID_PARAMS, "topic should be a string")
 	}
 
-	hexAssetID, err := common.HexStringToBytes(str)
-	if err != nil {
-		return respPacking(INVALID_PARAMS, err.Error())
+	var bucket float64
+	if _, ok := params["bucket"]; ok {
+		bucket, ok = params["bucket"].(float64)
+		if !ok {
+			return respPacking(INVALID_PARAMS, "bucket should be a float64")
+		}
 	}
 
-	assetID, err := common.Uint256ParseFromBytes(hexAssetID)
-	if err != nil {
-		return respPacking(INVALID_PARAMS, err.Error())
+	var offset float64
+	if _, ok := params["offset"]; ok {
+		bucket, ok = params["offset"].(float64)
```

Comment:
this should be `offset, ok = ...`

Reply:
actually, no, because then you won't be able to access `offset` outside of that `if`

Reply:
I think you missed the point, you are modifying `bucket` variable (instead of `offset`) using `params["offset"]`, same for below

Reply:
oh, I see, thanks